### PR TITLE
postgres preflight check  finalization 

### PR DESCRIPTION
**Pull request by marcparadise**
This PR finalizes the postgresql preflight checks after another round of testing on RDS, and replaces  the TODO urls for help/documentation. 

* bug fixed which prevented CSPG001 from being detected
* Updated URLs to final permanent values for all CSPG??? errors. 
* made postgresql version check a minimum version check, so instead of exactly 9.2 it now looks for 9.x where x >=2
* added new error CSPG017 that occurs when  required roles already exist in the external database.

### CODE REVIEW

**Comment by marcparadise** [comment]
ping @chef/lob

File: omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_checks.rb [hunk 1]
```diff
@@ -38,6 +38,20 @@ def backend?
                   node['enterprise']['name'] =>  PrivateChef }
     EnterpriseChef::Helpers.backend? faux_node
   end
+
+  # Helper function that let get the correct top-level value for a service
+  # node entry, givent that we haven't yet merged PrivateChef
```
Comment (tylercloke) [inline omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_checks.rb]: givent -> given

File: omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_checks.rb [hunk 2]
```diff
@@ -38,6 +38,20 @@ def backend?
...
+  def service_key_value(service_name, key)
+    # Ugh: config key munging pain redux
+    alt_service_name = (service_name == 'opscode-erchef' ? 'opscode_erchef' : service_name)
+    if PrivateChef.has_key?(alt_service_name)
+      if PrivateChef[alt_service_name].has_key?(key)
+        return PrivateChef[alt_service_name][key]
+      end
+    end
+    return node['private_chef'][service_name][key]
+  end
```
Comment (tylercloke) [inline omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_checks.rb]: That we need this method makes me sad, but if it works, it works.

**Comment by tylercloke** [comment]
LGTM, let's merge it after we finish our testing though.

File: omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_checks.rb (hunk 2 above)
Comment (marcparadise) [inline omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_checks.rb]: Yeah, I'm not a fan of this one at all.  I've lost count of the number of times - -> _ auto-mapping has bitten me - would love to just change our internal use to 'opscode_erchef'. Unfortunately  I think even that runs the risk of breaking user customer configurations if they've gone off into undocumented territory.

Comment (tylercloke) [inline omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_checks.rb]: :+1: Not criticizing the code, just the fact that we need it in the first place lol.

**Comment by tylercloke** [comment]
^ build http://wilson.ci.chef.co/job/chef-server-12-trigger-ad_hoc/667/downstreambuildview/

**Comment by tylercloke** [comment]
http://wilson.ci.chef.co/job/chef-server-12-trigger-ad_hoc/668/downstreambuildview/

**Comment by tylercloke** [comment]
Upgrade testing passed. Merging in 30 mins.

**Pull request merged**